pyproject.toml

```
@@ -0,0 +1,2 @@
+[build-system]
+requires = ['setuptools', 'cython <3']
```

Instructions you (correctly) deleted above said `cython >= .21`. If the `< 3` constraint is also correct (I don't know what pyscipopt needs so I have no opinion other than that the old instructions don't mention it) then you probably want
```suggestion
requires = ['setuptools', 'cython <3, >=0.21']
```

---

Yes, adding this is also fine and correct; but setting this lower bound is not really needed – these build system requirements are resolved in their own venv, and there won't be a conflict with other packages that would force a lower cython version into this venv.

Cython < 3 protects against the untested major Cython release that has been in the making for many years.